### ISSUE TITLE
Use of mosaic in the augmentation strategy

### ISSUE THREAD

**Issue by Luugaaa** [ivadomed/model_seg_sc-gm-lesion_human_ms_exvivo_t2star#30]
# Context

In Object detection, the mosaic augmentation is known to greatly improve the results, for this reason we tried it on our MRI segmentation project. 

# Method 

The mosaic is computed by taking 4 images in the batch and by sticking them together, the final image is 2x2 times bigger than a normal MRI image.

During inference, if there are less than 4 accessible images, an image is repeated and only the first mask of the repetition is considered. 

# Results 

Note: the following results are obtained by averaging over five folds the dice score obtained on validation set of the "full" 12 subjects dataset. 

The colors are : 
- **lime:** models trained with the mosaic augmentation
- **red:** base model, no mosaic augmentation

# Results

Note: I accidentally stopped the runs toward the end, however the results are still pretty clear. 

## Overall foreground dice results

<img width="735" height="436" alt="Image" src="https://github.com/user-attachments/assets/70b0de00-85f3-43b6-984f-8bc400770725" />

On the foreground classes, the dice score is almost identical with and without the mosaic augmentation. When we look at the detail of each class results, we can do the same observation. 

### Train Loss and Val Loss

Note: the loss is a combination of Dice (weighted at 1.0) and CE (weighted at 6.0)

**Val Loss**

<img width="735" height="436" alt="Image" src="https://github.com/user-attachments/assets/0f6a9097-0661-4b63-9228-07320875acbc" />

**Train loss** 

<img width="735" height="436" alt="Image" src="https://github.com/user-attachments/assets/215db9b9-440b-458c-b4f4-8579a68ccef9" />

**Observation:** In the case of the mosaic training, the Val loss and Train loss are closer, indicating that this augmentation is successfully preventing a certain level of overfitting. 

# Conclusion 

The mosaic augmentation is limiting overfitting. However, it's not improving the dice results on the validation set. Therefore, for simplicity we won't be using the mosaic augmentation from now on. Also, note that using 2x2 times bigger image requires more ressources when running inference: this can be limiting if the inference is done on a personal computer with a small gpu.
